pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2612,8 +2612,10 @@ func (ms *MutexStatus) LockAcquired(holderKey, lockKey string, currentHolders []
 }
 
 func (ms *MutexStatus) LockReleased(holderKey, lockKey string) bool {
-	i, _ := ms.GetHolding(lockKey)
-	if i >= 0 {
+	i, holder := ms.GetHolding(lockKey)
+	items := strings.Split(holderKey, "/")
+	holdingName := items[len(items)-1]
```

Comment:
What if `len(items) == 0`?

Reply:
Holderkey will have two default elements in synchronization scenario. Namespace/wf name. I can add a  check